src/ai/backend/storage/api/storages.py

```diff
@@ -0,0 +1,398 @@
+            content_type=object_info.content_type,
+            last_modified=object_info.last_modified,
+            etag=object_info.etag,
+        )
+        return web.Response(
+            status=HTTPStatus.OK,
+            body=dump_json_str(response.model_dump()),
+            content_type="application/json",
+        )
+
+    except Exception as e:
+        log.error(f"Get object info failed: {e}")
+        error_response = ErrorResponse(error=f"Get object info failed: {str(e)}")
+        return web.Response(
+            status=HTTPStatus.INTERNAL_SERVER_ERROR,
+            body=dump_json_str(error_response.model_dump()),
+            content_type="application/json",
+        )
+
+
+async def delete_object(request: web.Request) -> web.Response:
+    """Delete object"""
+    token_data = await validate_token_request(request)
+    if token_data.op != "delete":
+        raise web.HTTPBadRequest(reason="Invalid token operation for delete")
+
+    await log_client_api_entry(log, "delete_object", token_data)
+
+    try:
+        ctx: RootContext = request.app["ctx"]
+        storage_config = None
+        for storage in ctx.local_config.storages:
+            if storage.bucket == token_data.bucket:
+                storage_config = storage
+                break
+
+        if not storage_config:
+            raise web.HTTPBadRequest(
+                reason=f"No storage configuration found for bucket: {token_data.bucket}"
+            )
+
+        s3_client = S3Client(
+            bucket_name=token_data.bucket,
+            endpoint_url=storage_config.endpoint,
+            region_name=storage_config.region,
+            aws_access_key_id=storage_config.access_key,
+            aws_secret_access_key=storage_config.secret_key,
+        )
+
+        success = await s3_client.delete_object(token_data.key)
+
+        response = DeleteResponse(success=success)
+        return web.Response(
+            status=HTTPStatus.OK,
+            body=dump_json_str(response.model_dump()),
+            content_type="application/json",
+        )
+
+    except Exception as e:
+        log.error(f"Delete object failed: {e}")
+        error_response = ErrorResponse(error=f"Delete object failed: {str(e)}")
+        return web.Response(
+            status=HTTPStatus.INTERNAL_SERVER_ERROR,
+            body=dump_json_str(error_response.model_dump()),
+            content_type="application/json",
+        )
+
+
+def create_app(ctx: RootContext) -> web.Application:
+    app = web.Application()
+    app["ctx"] = ctx
+    app["prefix"] = "storages"
+
+    # Route definitions following the pattern from manager.py
+    app.router.add_route("POST", "/s3/upload", stream_upload)
+    app.router.add_route("POST", "/s3/presigned-upload-url", presigned_upload_url)
+    app.router.add_route("GET", "/s3/presigned-download-url", presigned_download_url)
+    app.router.add_route("GET", "/s3", get_object_info)
+    app.router.add_route("DELETE", "/s3", delete_object)
```

Comment:
It seems like additional information is needed to attach it only as an object.
It seems good to include bucket information in the URL path parameter overall.
`/v1/storages/s3/buckets/:bucket_name/files/upload`?

Reply:
Changed to `/v1/storages/s3/:storage_name/buckets/:bucket_name` format.